lib/ldp.js

```diff
@@ -184,7 +185,7 @@ class LDP {
     .then(result => callback(null, result), callback)
   }
 
-  post (hostname, containerPath, slug, stream, container, callback) {
+  post (host, containerPath, stream, { container, slug, extension }, callback) {
```

Comment:
Nitpick, but is there a reason to put only `container`, `slug`, and `extension` in the options object?

Reply:
Yes: the first three seem like elementary objects ("put this stream on this path on this host"), whereas the other are of lesser importance. Would that make sense?